project.go

```diff
@@ -188,28 +191,34 @@ func (project Project) WalkTodosOfDir(dirpath string, visit func(todo Todo) erro
 		return err
 	}
 
-	scanner := bufio.NewScanner(&outb)
+	g, ctx := errgroup.WithContext(context.Background())
+	// FIXME: Arbitrary max weight for preventing "too many open files" error
+	// This has to do with big projects and the limit of number of open file
+	// descriptors that a process may have at any given time.
+	sem := semaphore.NewWeighted(1024)
 
-	for scanner.Scan() {
+	for scanner := bufio.NewScanner(&outb); scanner.Scan(); {
 		filepath := scanner.Text()
 
 		stat, err := os.Stat(filepath)
 		if err != nil {
 			return err
 		}
-
-		if !stat.IsDir() {
-			err = project.WalkTodosOfFile(filepath, visit)
-			if err != nil {
-				return err
-			}
-		} else {
+		if stat.IsDir() {
 			// FIXME(#145): snitch should go inside of git submodules recursively
 			fmt.Printf("[WARN] `%s` is probably a submodule. Skipping it for now...\n", filepath)
+			continue
 		}
+
+		sem.Acquire(ctx, 1)
+		g.Go(func() error {
+			defer sem.Release(1)
+			// FIXME: Don't visit a path if it's other than source code (e.g. images, music)
```

Comment:
Why? Please see the description of the project. Snitch is supposed to be a language agnostic tool. Meaning that it doesn't really care what kind of files to work with. What's the actual problem here you are experiencing?

Reply:
Personally I'm not experiencing any problems. And I'm perfectly fine with the language agnostic aspect of this tool. I thought it would be a nice to have to ignore specific files (or directories).
If you think it's not in the scope of this project I will remove the fixme.

Reply:
If there is no problem then nothing to report. Please remove this FIXME.

Reply:
Done https://github.com/tsoding/snitch/pull/166/commits/3433938f6eee0811e92490a2fae11b1dc74deaa0!